### PR TITLE
Automated tests failing on Windows

### LINKED ISSUE #115: Automated tests failing on Windows
Windows adds an extra `\r` character to the WebServlet edit. Need a way to handle this character in our tests

See the failing WebServlet test here: https://github.com/eclipse/lsp4jakarta/blob/main/jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/servlet/JakartaServletTest.java#L60

```
Assert edit [0] expected:<WorkspaceEdit [
  changes = {}
  documentChanges = ArrayList (
    Either [
      left = TextDocumentEdit [
      textDocument = VersionedTextDocumentIdentifier [
        version = 0
        uri = "file:/C:/Users/Mrmat/Documents/_Winter%202021/LSP4Jakarta/lsp4jak
arta/jakarta-eclipse/org.eclipse.lsp4jakarta.tests/target/workingProjects/jakart
a-sample/src/main/java/io/openliberty/sample/jakarta/servlet/InvalidWebServlet.j
ava"
      ]
      edits = ArrayList (
        TextEdit [
          range = Range [
            start = Position [
              line = 9
              character = 0
            ]
            end = Position [
              line = 10
              character = 0
            ]
          ]
          newText = "@WebServlet(value = "")\n"
        ]
      )
    ]
      right = null
    ]
  )
  resourceChanges = null
]> but was:<WorkspaceEdit [
  changes = {}
  documentChanges = ArrayList (
    Either [
      left = TextDocumentEdit [
      textDocument = VersionedTextDocumentIdentifier [
        version = 0
        uri = "file:/C:/Users/Mrmat/Documents/_Winter%202021/LSP4Jakarta/lsp4jak
arta/jakarta-eclipse/org.eclipse.lsp4jakarta.tests/target/workingProjects/jakart
a-sample/src/main/java/io/openliberty/sample/jakarta/servlet/InvalidWebServlet.j
ava"
      ]
      edits = ArrayList (
        TextEdit [
          range = Range [
            start = Position [
              line = 9
              character = 0
            ]
            end = Position [
              line = 10
              character = 0
            ]
          ]
          newText = "@WebServlet(value = "")\r\n"
        ]
      )
    ]
      right = null
    ]
  )
  resourceChanges = null
]>
```

Automated tests for Windows can be enabled via GH Actions: 
https://github.com/eclipse/lsp4jakarta/blob/main/.github/workflows/lsp4jakarta.yaml#L9-L14

### PR DESCRIPTION
**Pull request by bensonnw**
Fix #115

### CODE REVIEW

**Comment by bensonnw** [comment]
@kathrynkodama, thx

**Comment by kathrynkodama** [comment]
@turkeylurkey please review as well

File: jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/core/JakartaForJavaAssert.java [hunk 1]
```diff
@@ -201,4 +220,8 @@ public static String fixURI(URI uri) {
         return uriString.replaceFirst("file:/([^/])", "file:///$1");
     }
 
+    // util. APIs
```
Comment (kathrynkodama) [inline jakarta.jdt/org.eclipse.lsp4jakarta.jdt.test/src/main/java/org/eclipse/lsp4jakarta/jdt/core/JakartaForJavaAssert.java:223]: Suggestion to add comment that the extra `\r` character appears on Windows

**Review by kathrynkodama — COMMENTED** [review]
Looks like there are failures in the automated GH actions build

**Review by kathrynkodama — COMMENTED** [review]
As part of this, update the GitHub workflow as well to run the automated tests on Windows. You should just have to update the os matrix to add windows: https://github.com/eclipse/lsp4jakarta/blob/79511d825462d9fb1e42e175145906c533caa0c4/.github/workflows/lsp4jakarta.yaml#L14

**Comment by kathrynkodama** [comment]
> Are the build failures caused by PR?

I manually re-ran the tests and look like they passed

**Comment by bensonnw** [comment]
hmm..., do we still need to add `windows-latest` to os?

**Comment by bensonnw** [comment]
A separate issue opened ( #311 ) for adding windows build option, if no further comments for this PR, please help to merge it. thx

File: .github/workflows/lsp4jakarta.yaml [hunk 2]
```diff
@@ -21,11 +21,11 @@ jobs:
           java-version: 11
       - name: Build LSP4Jakarta JDT LS Extension with Maven
         working-directory: ./jakarta.jdt
-        run: xvfb-run -a mvn -B install --file pom.xml
+		run xvfb-run -a mvn -B install --file pom.xml
```
Comment (turkeylurkey) [inline .github/workflows/lsp4jakarta.yaml:24]: Is it valid to change the spacing and remove the ":" ?

Comment (bensonnw) [reply .github/workflows/lsp4jakarta.yaml]: Fixed

**Review by turkeylurkey — APPROVED** [review]

**Review by kathrynkodama — APPROVED** [review]

**Pull request merged**